spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -0,0 +1,212 @@
+				: StreamSupport.stream(found.spliterator(), false)
+						.collect(Collectors.toList());
+	}
+
+	private StructuredQuery<Entity> getQuery(Object[] parameters) {
+		Builder<Entity> builder = StructuredQuery.newEntityQueryBuilder();
+
+		builder.setKind(this.datastorePersistentEntity.kindName());
+
+		if (this.tree.hasPredicate()) {
+			builder.setFilter(getFilter(parameters));
+		}
+
+		if (!this.tree.getSort().isUnsorted()) {
+			applySort(builder);
+		}
+
+		if (this.tree.isLimiting()) {
+			builder.setLimit(this.tree.getMaxResults());
+		}
+
+		return builder.build();
+	}
+
+	private void applySort(Builder builder) {
+		OrderBy[] orders = this.tree.getSort().get().map(sort -> {
+			String fieldName = ((DatastorePersistentProperty) this.datastorePersistentEntity
+					.getPersistentProperty(sort.getProperty())).getFieldName();
+			return sort.isAscending() ? OrderBy.asc(fieldName) : OrderBy.desc(fieldName);
+		}).toArray(OrderBy[]::new);
+		if (orders.length > 1) {
+			builder.setOrderBy(orders[0], Arrays.copyOfRange(orders, 1, orders.length));
+		}
+		else {
+			builder.setOrderBy(orders[0]);
+		}
+	}
+
+	private Filter getFilter(Object[] parameters) {
+		Iterator it = Arrays.asList(parameters).iterator();
+		Filter[] filters = this.filterParts.stream().map(part -> {
+			Filter filter;
+			String fieldName = ((DatastorePersistentProperty) this.datastorePersistentEntity
+					.getPersistentProperty(part.getProperty().getSegment()))
+							.getFieldName();
+			try {
+				switch (part.getType()) {
+				case IS_NULL:
+					filter = PropertyFilter.isNull(fieldName);
+					break;
+				case IS_EMPTY:
+					filter = PropertyFilter.isNull(fieldName);
+					break;
+				case SIMPLE_PROPERTY:
+					filter = PropertyFilter.eq(fieldName,
+							DatastoreNativeTypes.wrapValue(it.next()));
+					break;
+				case GREATER_THAN_EQUAL:
+					filter = PropertyFilter.ge(fieldName,
+							DatastoreNativeTypes.wrapValue(it.next()));
+					break;
+				case GREATER_THAN:
+					filter = PropertyFilter.gt(fieldName,
+							DatastoreNativeTypes.wrapValue(it.next()));
+					break;
+				case LESS_THAN_EQUAL:
+					filter = PropertyFilter.le(fieldName,
+							DatastoreNativeTypes.wrapValue(it.next()));
+					break;
+				case LESS_THAN:
+					filter = PropertyFilter.lt(fieldName,
+							DatastoreNativeTypes.wrapValue(it.next()));
+					break;
+				default:
+					throw new DatastoreDataException(
+							"Only equals, greater-than-or-equals, greater-than, less-than-or-equals, "
+									+ "less-than, and is-null are supported filters in Cloud Datastore.");
+
+				}
+				return filter;
+			}
+			catch (NoSuchElementException e) {
+				throw new DatastoreDataException(
+						"Too few parameters are provided for query method: "
+								+ this.queryMethod.getName());
+			}
+		}).toArray(Filter[]::new);
+		return filters.length > 1
+				? CompositeFilter.and(filters[0],
+						Arrays.copyOfRange(filters, 1, filters.length))
```

Comment:
See if we can fix this with `addFilter`.

Reply:
The method used, CompositeFilter.and(...), is the only way to create this set of AND filters.